RELEASE.md

```diff
@@ -0,0 +1,304 @@
+# Release process
+
+The release process starts with one or more tagged release candidates,
+when no more blocking issues needs to be fixed a final tag is created
+and the final release is rolled.
+
+confluent-kafka-dotnet uses semver versioning and loosely follows
+librdkafka's version, e.g. v0.11.x for the final release and
+v0.11.x-RC3 for the 3rd v0.11.x release candidate.
+
+With the addition of prebuilt binary packages we make use of travis-ci.org.
+These artifacts are downloaded from artifacts url of each ci build (for 
+example: https://ci.appveyor.com/project/ConfluentClientEngineering/confluent-kafka-dotnet/builds/39236976/artifacts),
+and then uploaded manually to https://www.nuget.org/.
+
+**Note**: Dotnet package versions use the same tag with librdkafka.
+          That is to say that while the librdkafka RC is named `v0.11.x-RC3`,
+          a dotnet client RC with the same version is also named `v0.11.x-RC3`.
+
+
+The following guide uses `v0.11.x-RC1` as version for a release candidate,
+replace as necessary with your version or remove `-RC..` suffix for the
+final release.
+
+To give you an idea what this process looks like we have this fabricated
+time line:
+
+ * Time to prepare for a release.
+ * A librdkafka release candidate is available.
+ * Check if any open PRs needs merging, do that.
+ * Feature freeze - don't merge any more PRs.
+ * Create a release candidate branch.
```

Comment:
"Create a release branch from the master branch ("MAJOR.MINOR.x") if this release has a new major or minor version from the previous release."

this should go after "Wait for librdkafka.redist ... "